### DISCUSSION TITLE
Fitting/Regression routines

### DISCUSSION THREAD

**Discussion by Onandon11** [pyvista/pyvista#3184] [Ideas]
To begin with; thank you guys for this great library, I still remember the time of writing my own wrappers around VTK which was quite tedious. PyVista replaced all of that and now it's a delight to work with.

However a major drawback for me still is the lack of fitting/regression routine. I know VTK is a Visualization library, however there a many-many functions and classes added which have nothing to do with visualization at all.

What do you guys think about adding routines for fitting/regression for numerous primitives? For example, Sphere's, Toroids, Cone's and maybe even 2D primitives like Circles and Lines?

**Comment by akaszynski** [comment]
> What do you guys think about adding routines for fitting/regression for numerous primitives? For example, Sphere's, Toroids, Cone's and maybe even 2D primitives like Circles and Lines?

I'm totally fine if you wanted to add in routines for that. PyVista could support these routines, but I think you would have better luck creating a library that depended on PyVista and then added the functionality there. We could add it here, but it would require a whole mess of unit testing and documentation. If you're up to that, feel free to add it.

**Reply by MatthewFlamm** [reply]
> I'm totally fine if you wanted to add in routines for that. PyVista could support these routines, but I think you would have better luck creating a library that depended on PyVista and then added the functionality there. We could add it here, but it would require a whole mess of unit testing and documentation. If you're up to that, feel free to add it.

I agree with this.  This package sounds like it will require new dependencies for optimization functionality.  IMO it would be best to keep packages like this separate, but possibly in the same family with similar naming.  `pyvista-regression` for example.
